generator/templates/base_struct_function.h

```diff
@@ -0,0 +1,50 @@
+{#- To avoid code duplication was crated this parent file, which contain common part used in:
+    "templates/functions/template.h" and "templates/structs/template.h". -#}
+{% include 'copyright.txt' %}
+{% block imports %}
+{%- for import in imports.enum %}
+#import "{{import}}.h"
+{%- endfor %}
+{%- if imports.struct %}
+{% endif -%}
+{%- for import in imports.struct %}
+@class {{import}};
+{%- endfor %}
+{%- endblock %}
+
+NS_ASSUME_NONNULL_BEGIN
+{% include 'description.jinja' %}
+@interface {{name}} : {{extends_class}}{{ending}}
```

Comment:
When creating an `init` that uses a deprecated type (for example, `SDLSeatControlData` which uses the deprecated `SDLSupportedSeat`) we need to wrap the `init` in

```
#pragma clang diagnostic push
#pragma clang diagnostic ignored "-Wdeprecated-declarations"

#pragma clang diagnostic pop
```

Reply:
done in https://github.com/smartdevicelink/sdl_ios/pull/1556/commits/97fc74359dc0ac2b615be9fb615416edb22becd9